packages/design-system-react-native/src/components/TextButton/TextButton.tsx

```diff
@@ -145,10 +144,8 @@ const TextButton = ({
   const nonTextOffSet = 2.5;
 
   // Get the final sizes of Icons to help with manual alignments
-  const finalStartIconSize =
-    startIconProps?.size || DEFAULT_TEXTBUTTON_PROPS.startIconProps.size;
-  const finalEndIconSize =
-    endIconProps?.size || DEFAULT_TEXTBUTTON_PROPS.endIconProps.size;
+  const finalStartIconSize = finalStartIconProps.size;
+  const finalEndIconSize = finalEndIconProps.size;
 
```

Comment:
non-blocking: Not sure if this was intended on being included but I think it's fine